contracts/RasinoRaffle.sol

```diff
@@ -46,42 +48,50 @@ contract RasinoRaffle is Ownable {
     /**
      * @dev Constructor
      */
-    constructor (address token, address devAddress, string memory _description) public {
+    constructor (address token, address devAddress, address trustFundAddress, string memory _description) public {
         _token = IERC20(token);
         description = _description;
         _devAddress = devAddress;
+        _trustFundAddress = trustFundAddress;
     }
 
     /**
      * @dev Sets both dev and ticket fees
      */
-    function setFees(uint256 ticketFee, uint256 devFee) external onlyOwner{
+    function setFees(uint256 ticketFee, uint256 devFee, uint256 trustFundFee) external onlyOwner{
         _ticketFee = ticketFee;
         _devFee = devFee;
+        _trustFundFee = trustFundFee;
     }
 
+    /**
+     * @dev Sets the addresses in contract (by owner only)
+     */
+    function setAddresses(address devAddress, address trustFundAddress) external onlyOwner{
+        _devAddress = devAddress;
+        _trustFundAddress = trustFundAddress;
+    }
     /**
      * @dev Claim earnings for sender
      */
     function claimEarnings() external {
         address from = _msgSender();
+        require(canCollectEarnings());
         _claimEarnings(from);
     }
     /**
      * @dev Claim earnings for ticket holders
      */
     function _claimEarnings(address account) internal {
+        require(userMemo[account].isValue);
+        require(userMemo[account].round < rounds.length); // must be a past round
+        require(rounds[userMemo[account].round].collected);
         uint earnings = _estimateEarnings(account);
         require(_token.transfer(account, earnings));
         delete userMemo[account];
     }
 
     function _estimateEarnings(address account) internal view returns(uint) {
-        
-        require(userMemo[account].isValue);
```

Comment:
Just realized this may still be wanted/needed. could cause a revert... I should also create tests for that

Reply:
I resolved by adding a check in the public facing version